### DISCUSSION TITLE
"View Full Record" not working

### DISCUSSION THREAD

**Discussion by rflore2** [CollectionBuilder/collectionbuilder.github.io#129] [Q&A] [answered]
Hi all,

I'm testing out CollectionBuilder to see if it will work for a library database project: [https://rflore2.github.io/ReadingRageDemo/](https://rflore2.github.io/ReadingRageDemo/)

I've encountered a problem: When I click on "View Full Record" for an object in browse search ([https://rflore2.github.io/ReadingRageDemo/browse.html] https://rflore2.github.io/ReadingRageDemo/browse.html), the object doesn't load. It takes me to a new page, and a circle just spins in the middle of the page indefinitely. Objects are viewable in a table ([https://rflore2.github.io/ReadingRageDemo/data.html](https://rflore2.github.io/ReadingRageDemo/data.html)) but I have the same issue when I click on the title to see an individual object. All the other features seem to be working and are pulling data from these objects, I just can't view them individually.

On my pages build and deployment, I see this warning: "The github-pages gem can't satisfy your Gemfile's dependencies. If you want to use a different Jekyll version or need additional dependencies, consider building Jekyll site with GitHub Actions: https://jekyllrb.com/docs/continuous-integration/github-actions/"

Does anyone know what the problem is and how to troubleshoot it?

**Comment by evanwill** [comment, marked as answer]
@rflore2 the issue you are encountering on the Item page is that your metadata does not contain a "format" field which is required to function in CB-GH. Check the details: https://collectionbuilder.github.io/cb-docs/docs/metadata/gh_metadata/#format

(a longer explanation--> on CB-GH all the pages have a fall back if the item in the metadata is missing a format, EXCEPT the Item page which needs to choose the correct representation of the Item to display based on the "format". To debug a page that won't load, you can open your browser dev tools, and check the console to see what error is coming up in the javascript. In this case it will say `Uncaught TypeError: record.format is null`, meaning it ran into an item with no format and couldn't finish creating the item page. Ultimately, we (at CB) should add a fall back there too to avoid the issue--but it also sometimes harder to debug why your item page isn't showing as you except if there is no visible error... )

**Reply by evanwill** [reply]
Secondly, on the gh-pages warning--> you can just ignore that--it is building fine!

We added some gems to the Gemfile to make it possible to build on Ruby 3 on your local machine. GH-Pages uses an old version of ruby and jekyll, so it gives that warning, but will go on to build just fine.

**Reply by rflore2** [reply]
Thank you so much for this explanation! I've added the format field and all is well now.
